pkg/controller/instance/instance_controller.go

```diff
@@ -97,7 +97,10 @@ func (r *Reconciler) SetupWithManager(
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&kudov1beta1.Instance{}).
+		// Owns() is equivalent to Watches(&source.Kind{Type: <ForType-apiType>}, &handler.EnqueueRequestForOwner{OwnerType: apiType, IsController: true})
 		Owns(&kudov1beta1.Instance{}).
+		// the Watches() below is almost the same as Owns() above, but with IsController: false for reconciliation of parent Instances
+		Watches(&source.Kind{Type: &kudov1beta1.Instance{}}, &handler.EnqueueRequestForOwner{OwnerType: &kudov1beta1.Instance{}, IsController: false}).
```

Comment:
Can you describe in the comment that we do this to reconcile dependencies? I.e. watch with `IsController: true` as well as with `IsController: false`.

Reply:
agree with nfnt and what's confusing here is the kind for `Owns` is the same kind for `Watches`.  It looks like it is registered twice... once with controller == true and once with controller == false.  The comments explain the what... they don't explain the why.

Reply:
I added more comments 👍